### DISCUSSION TITLE
In built terminal inside the editor

### DISCUSSION THREAD

**Discussion by brvank** [godotengine/godot-proposals#6210] [Editor]
What about a terminal inside the editor like those inside vscode, android studio, eclipse and other IDEs.
I think it would be helpful for using vcs commands inside the frame of reference of current project.

**Comment by aXu-AP** [comment]
I have been thinking about something like this also. Would be helpful for testing various functions without having to write tool scripts and reloading scenes.
My idea is that you could write gdscript code, which would be executed immediately by the root node of the currently opened scene. Maybe use `shift+enter` to enter multiline code.

**Comment by dalexeev** [comment]
See REPL:

* #6171
* #4473
* godotengine/godot#60134

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/1379 (see also the proposals linked above).
